src/models/multiBarChart.js

```diff
@@ -179,11 +179,6 @@ nv.models.multiBarChart = function() {
             if (showLegend) {
                 legend.width(availableWidth - controlWidth());
 
-                if (multibar.barColor())
-                    data.forEach(function(series,i) {
-                        series.color = d3.rgb('#ccc').darker(i * 1.5).toString();
-                    });
```

Comment:
In the case of barColor() being used the legend should be shades of gray... this is because the legend represents each series, and when you set barColor() you're abandoning colors per series and instead making each series a particular shading... that's why it was shades of gray before.